### DISCUSSION TITLE
Order of data in variable of a dataset when using open_mfdataset

### DISCUSSION THREAD

**Discussion by PythonSchlumpf** [pydata/xarray#5942] [Q&A]
Dear xarray-users,

I understand now that with `xr.open_mfdataset` I can open multiple hdf5 files and store the data in a dataset. My hdf5 files don't have labeled dimensions. I get a dataset with 
```py
Dimensions:  (phony_dim_0: 21, phony_dim_1: 2167, phony_dim_2: 2070)
Dimensions without coordinates: phony_dim_0, phony_dim_1, phony_dim_2
Data variables:
    data     (phony_dim_0, phony_dim_1, phony_dim_2) uint32 dask.array<chunksize=(10, 2167, 2070), meta=np.ndarray
```
I used:
```
ds = xr.open_mfdataset("path/to/data_*.h5", combine='nested', concat_dim='phony_dim_0', group=sub_group)

```
to receive the output at the top.

When I open the hdf5 files, I have 10,10,1 2D numpy arrays, hence `(phony_dim_0: 21)`. But how can I be sure, the 21 2D arrays are in the right order, please? How does `xr.open_mfdataset` keeps the correct order when putting all arrays in one big one together, please? 

And my second question would be, is there a way to collect the attributes of the different hdf5 wile using `open_mfdataset` and store them as well with the dataset? I have per hdf5 file one attribute with two values.

Thank you. 
PS: Sorry, if these questions are too easy.

**Comment by TomNicholas** [comment]
> How does xr.open_mfdataset keeps the correct order when putting all arrays in one big one together, please?

With `combine='nested'`, `open_mfdataset` will arrange the files in the order you supply them. Try it with a list of just a couple of files.

> And my second question would be, is there a way to collect the attributes of the different hdf5 wile using open_mfdataset and store them as well with the dataset?

Not really - consistent rules for combining attributes from multiple datasets is a difficult problem in general, and at the moment `open_mfdataset` just lets you choose one "master" file to extract the attributes from.

>  I have per hdf5 file one attribute with two values.

What I suggest you try is using the `preprocess` function to promote these attributes to data variables, so that `combine_nested` within `open_mfdataset` will concatenate them all together along with the other data in the dataset.

**Reply by keewis** [reply]
> Not really

that's not *exactly* true, `open_mfdataset` has a `combine_attrs` option that allows customizing what it does with the `attrs` (you can even pass your own function if you're not content with the built-in strategies). It seems I forgot to add that to the docstring in #4971, though.

**Reply by PythonSchlumpf** [reply]
Keewis and TomNicholas, many thanks. Do you have by any chance some code snippets as an example, please?
